pybossa/api/__init__.py

```diff
@@ -713,3 +713,36 @@ def _get_valid_service(task_id, service_name, payload, proxy_service_config):
         'Task {} loaded for user {} failed calling {} service with payload {}'.format(task_id, current_user.id, service_name, payload))
 
     return abort(403, 'The request data failed validation')
+
+
+@jsonpify
+@login_required
+@csrf.exempt
+@blueprint.route('/task/<int:task_id>/assign', methods=['POST'])
+@ratelimit(limit=ratelimits.get('LIMIT'), per=ratelimits.get('PER'))
+def assign_task(task_id=None):
+    """Assign task to users for locked, user_pref and task_queue schedulers."""
+    if not current_user.is_authenticated:
```

Comment:
1. This can be removed as login_required is in place.
2. Add check for user has obtained the lock before assigning the user.

Reply:
Makes sense. Thanks!